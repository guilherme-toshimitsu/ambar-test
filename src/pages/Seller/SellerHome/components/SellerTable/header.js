import React from 'react';

const headers = [
  {
    id: 'sellerID',
    label: 'Seller Id',
  },
  {
    id: 'NEIssued',
    label: 'NE Emitida',
    tooltipText: (
      <span>
        Dias corridos para liberação de recebíveis a partir do status <strong> Nota fiscal emitida </strong>, quando o
        pedido não for Magalu Entregas.
      </span>
    ),
  },
  {
    id: 'NEIssuedMle',
    label: 'NF Emitida MLE',
    tooltipText: (
      <span>
        Dias corridos para liberação de recebíveis a partir do status <strong> Nota fiscal emitida </strong>, quando o
        pedido for Magalu Entregas.
      </span>
    ),
  },
  {
    id: 'shipment',
    label: 'Despachado',
    tooltipText: (
      <span>
        Dias corridos para liberação de recebíveis a partir do status <strong> Despachado </strong>, quando o pedido não
        for Magalu Entregas.
      </span>
    ),
  },
  {
    id: 'shipmentMLE',
    label: 'Despachado MLE',
    tooltipText: (
      <span>
        Dias corridos para liberação de recebíveis a partir do status <strong> Nota fiscal emitida </strong>, quando o
        pedido for Magalu Entregas.
      </span>
    ),
  },
  {
    id: 'delivered',
    label: 'Entregue',
    tooltipText: (
      <span>
        Dias corridos para liberação de recebíveis a partir do status <strong> Nota fiscal emitida </strong>, quando o
        pedido não for Magalu Entregas.
      </span>
    ),
  },
  {
    id: 'actions',
    // label: 'Ações',
    label: ' ',
  },
];

export { headers };
