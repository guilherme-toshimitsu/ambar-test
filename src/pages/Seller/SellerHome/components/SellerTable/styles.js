const styles = {
  firstHeaderCell: {
    backgroundColor: '#f4f4f4',
    paddingLeft: 32,
    color: 'rgba(0, 0, 0, 0.87)',
    fontWeight: 500,
    lineHeight: '1.5rem',
    minWidth: 232,
    height: 50,
  },
  headerCell: {
    backgroundColor: '#f4f4f4',
    textAlign: 'center',
    minWidth: 48,
    height: 50,
  },
  loadingIconContainer: {
    width: '100%',
    textAlign: 'center',
    padding: '30px 0',
    marginLeft: 10,
  },
  firstBodyTableCell: {
    paddingLeft: 32,
    height: 60,
  },
  bodyTableCell: {
    textAlign: 'center',
    // width: 78,
    fontSize: 13,
    color: '#404040',
    height: 60,
  },
};

export default styles;
