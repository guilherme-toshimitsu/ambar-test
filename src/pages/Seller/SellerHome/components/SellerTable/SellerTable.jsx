import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';

import { TableBody, TableCell, TableHead, TableRow, CircularProgress } from '@material-ui/core';

import { InfoIconTooltip } from '@components/Tooltip';
import ControlledLoading from '@components/ControlledLoading';
import NotFound from '@components/NotFound';

import { headers } from './header';

import styles from './styles';

const useStyles = makeStyles(styles);

const checkArray = data => data && typeof data === 'object' && data.length;

const SellerTable = ({ data, isLoading }) => {
  const classes = useStyles();
  return (
    <>
      <TableHead>
        <TableRow>
          {headers.map((header, idx) => (
            <TableCell key={`${header.id}${idx}`} className={idx === 0 ? classes.firstHeaderCell : classes.headerCell}>
              {header.label}
              {!!header.tooltipText && <InfoIconTooltip title={header.tooltipText} />}
            </TableCell>
          ))}
        </TableRow>
      </TableHead>
      <TableBody>
        <ControlledLoading
          isLoading={isLoading}
          loadingComponent={
            <TableRow>
              <TableCell colSpan={headers.length}>
                <div className={classes.loadingIconContainer}>
                  <CircularProgress />
                </div>
              </TableCell>
            </TableRow>
          }
          renderComponent={
            <>
              {checkArray(data) ? (
                data.map((row, idx) => (
                  <TableRow hover role="checkbox" tabIndex={-1} key={`row${idx}`}>
                    {headers.map((header, idx) => (
                      <TableCell
                        key={`${idx}${row[header.id]}`}
                        className={idx === 0 ? classes.firstBodyTableCell : classes.bodyTableCell}
                      >
                        {idx === 0 ? <a href={`/seller/${row[header.id]}`}>{row[header.id]}</a> : row[header.id]}
                      </TableCell>
                    ))}
                  </TableRow>
                ))
              ) : (
                <TableRow>
                  <TableCell colSpan={headers.length}>
                    <div className={classes.loadingIconContainer}>
                      <NotFound />
                    </div>
                  </TableCell>
                </TableRow>
              )}
            </>
          }
        />
      </TableBody>
    </>
  );
};

SellerTable.defaultProps = {
  isLoading: false,
  data: [],
};

SellerTable.propTypes = {
  isLoading: PropTypes.bool,
  data: PropTypes.array,
};

export default SellerTable;
