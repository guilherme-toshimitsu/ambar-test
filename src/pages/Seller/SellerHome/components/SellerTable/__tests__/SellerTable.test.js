import React from 'react';
import { render } from '@testing-library/react';
import SellerTable from '../';

describe('<SellerTable> SellerTable', () => {
  it('renders', () => {
    const container = render(<SellerTable />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
