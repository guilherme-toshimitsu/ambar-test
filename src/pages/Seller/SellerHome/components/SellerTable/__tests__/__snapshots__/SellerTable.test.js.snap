// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellerTable> SellerTable renders 1`] = `undefined`;
